Add security guide for web-based projects

zallarak opened the pull request:
Simple checks could have prevented the MyEtherWallet DNS hack, and the BlackWallet client-side script injection; this guide covers them and more.

As more projects are released, we should publish basic guidelines that outline basic web-sec standards for people building Stellar apps. We shouldn't be the authority on this, but we should cover important concepts for Stellar-app developers to ensure our ecosystem stays healthy.

theaeolianmachine commented:
@stanford-scs, would you mind giving a look over this doc - is there anything else we'd like to have covered? Happy to go over in person.

theaeolianmachine commented:
Thanks for sending this over @zallarak!

zallarak commented:
Just a friendly bump on this, cc @theaeolianmachine.

zallarak commented:
@stanford-scs thanks for your **very** valuable feedback! Will make an iteration that addresses your comments.

tomquisel commented:
@zallarak We're looking forward to your iteration!

theaeolianmachine commented:
Hey @zallarak just checking in here!

zallarak commented:
Sorry for the delay! I'm back on this. I updated per comments.

@stanford-scs I incorporated all of the suggested edits. Regarding the question-list you left; I think many of those issues have many ways of being solved. However, my intent in making this guide was to take projects from zero security to a baseline that is "acceptable", since anyone can technically build on Stellar. I think it still adds value in its current form without getting too specific. I am open to adding deep-dives separately for the wallet issues you listed; does that sound OK to you?

Again, pardon my delay, promise to be quicker on this.

ire-and-curses commented:
+1 for getting this out sooner and then doing something deeper. This is already a *ton* better than the present situation (nada).

theaeolianmachine commented on guides/walkthroughs/securing-web-projects.md:
```suggestion
Hash passwords with a time-tested scheme. The winner of the last password hashing contest was Argon2. Balloon Hashing is also worth looking into.
```

zallarak commented:
@theaeolianmachine great catch. I also reformatted to 100 char width - good call.

theaeolianmachine commented:
Thanks again @zallarak!
